warehouse/router/upload_stats.go

```diff
@@ -69,6 +69,17 @@ func (job *UploadJob) generateUploadSuccessMetrics() {
 		return
 	}
 
+	eventTimeRanges, err := job.stagingFileRepo.GetEventTimeRangesByUploadID(job.ctx, job.upload.ID)
+	if err != nil {
+		job.logger.Warnn("event time ranges for upload", obskit.Error(err))
+		return
+	}
+
+	for _, eventTimeRange := range eventTimeRanges {
+		job.stats.eventDeliveryTime.Since(eventTimeRange.FirstEventAt)
+		job.stats.eventDeliveryTime.Since(eventTimeRange.LastEventAt)
+	}
```

Comment:
We are not capturing the events distribution in the staging files here. I think we should use `total_events` from `wh_staging_files` table here.

Reply:
We currently don't have a way to capture this. There is a hacky way to have a for-loop from 0 to total_events, but I would not prefer it as total_events can be huge.